js/reviewApi.js

```diff
@@ -76,87 +92,91 @@ function createNewUser(review = null, reviewCallback = null) {
 		"phone": $('[name="tel"]').val()
 	};
 
-	console.log('user object', userObj);
-
 	userSettings = {
 		'url': '/user/new',
 		'method': 'POST',
 		'data': JSON.stringify(userObj)
 	};
 
-	console.log('user settings', userSettings);
-
-	$.ajax(userSettings)
-		.done(function (response) {
-			if (review !== null) {
-				review.id = response.id;
-				reviewCallback(review.id);
-			}
-			return response.id;
-		})
-		.fail(function (response) {
-			// FIXME let's fail gracefully
-			alert("broke it");
-			console.log(response);
-		})
+	return $.ajax(userSettings);
 };
 
 $(document).ready(function () {
 	// get all review and display
 	getAllReviews();
 
+	// add a submit event handler on th review form
 	$("#reviewForm").submit(function (e) {
```

Comment:
What is the 'e' within the function parenthesis for?

Reply:
I also would like to know.

Reply:
> What is the 'e' within the function parenthesis for?

"The e parameter of the function is an optional parameter of the input event handler which equals to a JavaScript Event object that contains information regarding what action or event just happened. When the handleUserInput() function is invoked, the action that occurs, like a change in the input field, is passed as the value of e. Then, the information about the action can be accessed, such as e.target.value, which would be the value of the text input element." I found that on codeacademy https://discuss.codecademy.com/t/what-is-the-e-parameter-in-the-input-event-handler-function/401998

I would not be able to explain it better lol.

Reply:
That is correct. So we could do something like:

```JS
var event = $('#reviewForm').submit()
```

But since JS is a functional programming language the better way to do it is to use the anonymous function capability and pass the return of `submit()` into said function as an arguement.

TL;DR `e` is the return from submit which is the actual event object